cpp/src/io/utilities/column_buffer.cpp

```diff
@@ -56,11 +56,12 @@ std::unique_ptr<column> gather_column_buffer::make_string_column_impl(rmm::cuda_
   return make_strings_column(*_strings, stream, _mr);
 }
 
-void cudf::io::detail::inline_column_buffer::allocate_strings_data(rmm::cuda_stream_view stream)
+void cudf::io::detail::inline_column_buffer::allocate_strings_data(bool memset_data,
+                                                                   rmm::cuda_stream_view stream)
 {
   CUDF_EXPECTS(type.id() == type_id::STRING, "allocate_strings_data called for non-string column");
   // size + 1 for final offset. _string_data will be initialized later.
-  _data = create_data(data_type{type_id::INT32}, size + 1, stream, _mr);
+  _data = create_data(data_type{type_id::INT32}, size + 1, memset_data, stream, _mr);
```

Comment:
Same question as below: https://github.com/rapidsai/cudf/pull/16281/files#r1699007201

Reply:
```suggestion
  _data = create_data(data_type{type_to_id<size_type>()}, size + 1, memset_data, stream, _mr);
```

Reply:
I believe this is resolved in this utility function
https://github.com/rapidsai/cudf/blob/af57286536fc21b47b80e45be222773b751600c9/cpp/src/io/utilities/column_buffer_strings.cu#L25
There is some efficiency builtin here where if the offsets fit, they do not need to be copied.
(Disclaimer: I'm not exactly how all of this works myself).